src/wallet/wallet.cpp

```diff
@@ -1695,8 +1695,9 @@ int64_t CWallet::RescanFromTime(int64_t startTime, const WalletRescanReserver& r
     }
 
     if (startBlock) {
-        const CBlockIndex* const failedBlock = ScanForWalletTransactions(startBlock, nullptr, reserver, update);
-        if (failedBlock) {
+        const CBlockIndex* failedBlock = nullptr;
+        // TODO: this should take into account failure by ScanResult::USER_ABORTED
```

Comment:
I'll address this todo in a follow-up.

Reply:
Follow-up is here, I'll wait to open until this is accepted:
https://github.com/Empact/bitcoin/tree/rescan-from-time

Reply:
#14942